File: lib/models/place_autocomplete_response.dart
class PlaceAutoCompleteResponse {
  final String placeName;
  final String roadAddress;
  final double latitude;
  final double longitude;

  PlaceAutoCompleteResponse({
    required this.placeName,
    required this.roadAddress,
    required this.latitude,
    required this.longitude,
  });

  factory PlaceAutoCompleteResponse.fromJson(Map<String, dynamic> json) {
    return PlaceAutoCompleteResponse(
      placeName: json['placeName'] ?? '',
      roadAddress: json['roadAddress'] ?? '',
      latitude: (json['latitude'] ?? 0).toDouble(),
      longitude: (json['longitude'] ?? 0).toDouble(),
    );
  }
}

File: backend/src/main/java/net/skhu/dto/PlaceAutoCompleteResponse.java
package net.skhu.dto;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PlaceAutoCompleteResponse {
    private String placeName;
    private String roadAddress;
    private double latitude;
    private double longitude;
}
